mod_prometheus.rs

```diff
@@ -316,36 +316,68 @@ fn prometheus_load(mod_int: &ModInterface) -> Status {
-            if last_bridge_hup_cause == "NORMAL_CLEARING" {
+
+        let mut callid = String::new();
+        let mut uniqueId = String::new();
+        let mut direction = String::new();
+
+        if let Some(sip_callid) = e.header("variable_sip_call_id"){
+            callid = sip_callid.to_string();
+        }
+        if let Some(unique_id )= e.header("Unique-ID") {
+            uniqueId = unique_id.to_string();
+        }
+        if let Some(call_direction) = e.header("Call-Direction") {
+            direction = call_direction.to_string();
+        }
+        fslog!(INFO, "callid:{:#?} uniqueId:{:#?} {:#?} CHANNEL_HANGUP_COMPLETE\n", callid, uniqueId, direction);
+
+        if let Some(hupCause) = e.header("Hangup-Cause") {
+
+            fslog!(INFO, "callid:{:#?} uniqueId:{:#?} {:#?} CHANNEL_HANGUP_COMPLETE hupCause:{:#?}\n", callid, uniqueId, direction, hupCause.clone());
+
+            if hupCause == "NORMAL_CLEARING" {  // NORMAL_CLEARING or ORIGINATOR_CANCEL or NO_USER_RESPONSE
                 if let Some(billsecvar) = e.header("variable_billsec") {
-                    let parsed_time = billsecvar.parse::<i64>();
+                    let parsed_time = billsecvar.parse::<u64>();
                     if parsed_time.is_ok() {
                         let bill_seconds = parsed_time.unwrap() ;
-                        if let Some(direction) = e.header("Call-Direction") {
-                            if direction == "outbound" {
-                                COUNTERS[FSCounter::SessionsOutboundCallDurationTotal].lock().unwrap().increment_by(bill_seconds as f64);
-                                COUNTERS[FSCounter::SessionsOutboundCallHangupComplete].lock().unwrap().increment();
-                                let acd = COUNTERS[FSCounter::SessionsOutboundCallDurationTotal].lock().unwrap().value() /
-                                                                COUNTERS[FSCounter::SessionsOutboundCallHangupComplete].lock().unwrap().value();
-                                GAUGES[FSGauge::SessionsOutboundACD].lock().unwrap().set(acd);
-                            } else if direction == "inbound" {
-                                COUNTERS[FSCounter::SessionsInboundCallDurationTotal].lock().unwrap().increment_by(bill_seconds as f64);
-                                COUNTERS[FSCounter::SessionsInboundCallHangupComplete].lock().unwrap().increment();
-                                let acd = COUNTERS[FSCounter::SessionsInboundCallDurationTotal].lock().unwrap().value() /
-                                                COUNTERS[FSCounter::SessionsInboundCallHangupComplete].lock().unwrap().value();
-                                GAUGES[FSGauge::SessionsInboundACD].lock().unwrap().set(acd);
-                            }
+                        if direction == "outbound" {
+                            COUNTERS[FSCounter::SessionsOutboundCallDurationTotal].lock().unwrap().increment_by(bill_seconds as f64);
```

Comment:
shouldn't it be u64 as well?

Reply:
no, this is how the function has been designed by Moises Sylva